### PR TITLE
WDCC:Support all Config Types

### PR DESCRIPTION
**Pull request by Lutzi3**
Before only config type 00 (general) was supported. Now all types are allowed. This enables ABAPGit for instance to transport (cross client) Fiori Catalogs and Groups !

### CODE REVIEW

File: src/objects/zcl_abapgit_object_wdcc.clas.abap [hunk 1]
```diff
@@ -24,7 +25,7 @@ CLASS ZCL_ABAPGIT_OBJECT_WDCC IMPLEMENTATION.
           ls_config_key TYPE wdy_config_key.
 
     ls_config_key-config_id = ms_item-obj_name+0(32).
-    ls_config_key-config_type = '00'.
+    ls_config_key-config_type = ms_item-obj_name+32(2).
```
Comment (mbtools) [inline src/objects/zcl_abapgit_object_wdcc.clas.abap:28]: ls_config_key-config_var = ms_item-obj_name+34(6).

File: src/objects/zcl_abapgit_object_wdcc.clas.abap [hunk 2]
```diff
@@ -46,7 +47,7 @@ CLASS ZCL_ABAPGIT_OBJECT_WDCC IMPLEMENTATION.
           lv_subrc      TYPE sysubrc.
 
     ls_config_key-config_id = ms_item-obj_name+0(32).
-    ls_config_key-config_type = '00'.
+    ls_config_key-config_type = ms_item-obj_name+32(2).
```
Comment (mbtools) [inline src/objects/zcl_abapgit_object_wdcc.clas.abap:50]: ls_config_key-config_var = ms_item-obj_name+34(6).

File: src/objects/zcl_abapgit_object_wdcc.clas.abap [hunk 3]
```diff
@@ -221,7 +236,7 @@ CLASS ZCL_ABAPGIT_OBJECT_WDCC IMPLEMENTATION.
           ls_config_key TYPE wdy_config_key.
 
     ls_config_key-config_id = ms_item-obj_name+0(32).
-    ls_config_key-config_type = '00'.
+    ls_config_key-config_type = ms_item-obj_name+32(2).
```
Comment (mbtools) [inline src/objects/zcl_abapgit_object_wdcc.clas.abap:239]: ls_config_key-config_var = ms_item-obj_name+34(6).

File: src/objects/zcl_abapgit_object_wdcc.clas.abap [hunk 4]
```diff
@@ -323,7 +338,7 @@ CLASS ZCL_ABAPGIT_OBJECT_WDCC IMPLEMENTATION.
                  ig_data =  ms_item-obj_name ).
 
     ls_config_key-config_id = ms_item-obj_name+0(32).
-    ls_config_key-config_type = '00'.
+    ls_config_key-config_type = ms_item-obj_name+32(2).
```
Comment (mbtools) [inline src/objects/zcl_abapgit_object_wdcc.clas.abap:341]: ls_config_key-config_var = ms_item-obj_name+34(6).

**Review by mbtools — CHANGES_REQUESTED** [review]
Thanks for the improvement. I looked at `TADIR` and there are cases where `obj_name` includes the third part of the key. We should add it to be complete.

**Comment by mbtools** [comment]
Example:

image

File: src/objects/zcl_abapgit_object_wdcc.clas.abap (hunk 2 above)
Comment (Lutzi3) [reply src/objects/zcl_abapgit_object_wdcc.clas.abap:51]: was added

File: src/objects/zcl_abapgit_object_wdcc.clas.abap (hunk 3 above)
Comment (Lutzi3) [reply src/objects/zcl_abapgit_object_wdcc.clas.abap:241]: was added

File: src/objects/zcl_abapgit_object_wdcc.clas.abap (hunk 4 above)
Comment (Lutzi3) [reply src/objects/zcl_abapgit_object_wdcc.clas.abap:344]: was added

**Review by mbtools — APPROVED** [review]

**Comment by mbtools** [comment]
Closes #4972

Could you please create a small test case for 07 and add it to https://github.com/abapGit-tests/WDCC_Fiori?

**Comment by Lutzi3** [comment]
> Closes #4972
> 
> Could you please create a small test case for 07 and add it to https://github.com/abapGit-tests/WDCC_Fiori?

This was done. I created another Pull request for this repo.

**Comment by larshp** [comment]
welcome to abapGit as a contributor 🎉

**Comment by Lutzi3** [comment]
> welcome to abapGit as a contributor 🎉

It is my pleasure ! 
In the context of Custom Code Migration (I am development architect and specialised on this topic) I use ABAPGit very often for transfers or archivings.
Kind Regards
Lutz

**Pull request merged**